geomesa-spark/geomesa-spark-jts/src/main/scala/org/locationtech/geomesa/spark/jts/udf/SpatialRelationFunctions.scala

```diff
@@ -129,14 +128,13 @@ object SpatialRelationFunctions {
 
   def fastDistance(c1: Coordinate, c2: Coordinate): Double = {
     val calc = geoCalcs.get()
-    calc.setStartingGeographicPoint(c1.x, c1.y)
-    calc.setDestinationGeographicPoint(c2.x, c2.y)
-    calc.getOrthodromicDistance
+    val startPoint = spatialContext.getShapeFactory.pointXY(c1.x, c1.y)
+    DistanceUtils.DEG_TO_KM * calc.distance(startPoint, c2.x, c2.y) * 1000
```

Comment:
might be worth checking to see if this class is thread-safe - if so, we don't need the threadlocal wrapper any more

Reply:
The [SpatialContext](https://locationtech.github.io/spatial4j/apidocs/com/spatial4j/core/context/SpatialContext.html) is thread safe, but I'm not sure if that still holds for the DistanceCalculator if multiple threads can call `getDistCalc`